app/services/mute_service.rb

```diff
@@ -1,17 +1,22 @@
 # frozen_string_literal: true
 
 class MuteService < BaseService
-  def call(account, target_account, notifications: nil)
+  def call(account, target_account, notifications: nil, duration: 0)
     return if account.id == target_account.id
 
-    mute = account.mute!(target_account, notifications: notifications)
+    mute = account.mute!(target_account, notifications: notifications, duration: duration)
 
     if mute.hide_notifications?
       BlockWorker.perform_async(account.id, target_account.id)
     else
       MuteWorker.perform_async(account.id, target_account.id)
     end
 
+    if duration != 0
+      jid = DeleteMuteWorker.perform_at(duration.seconds, account.id, target_account.id)
+      mute.update!(unmute_jid: jid)
+    end
```

Comment:
For scheduled stuff, we usually store them in the database to schedule within Sidekiq the tasks that have to be executed soon. This is a Postgres / Redis tradeoff. In this case, since I expect timed mutes to be somewhat rare, I think your approach makes more sense, but I'd like @Gargron to confirm.